src/main/scala/com/microsoft/partnercatalyst/fortis/spark/statistics/ComputedAnomalyKmeansModelParameter.scala

```diff
@@ -0,0 +1,112 @@
+package com.microsoft.partnercatalyst.fortis.spark.statistics
+
+import java.time.{ZoneId, ZonedDateTime}
+
+import com.microsoft.partnercatalyst.fortis.spark.analyzer.timeseries.PeriodType
+import com.microsoft.partnercatalyst.fortis.spark.dto.TopicCount
+
+case class ComputedAnomalyKmeansModelParameter(
+  key: String,
+  k: Int,
+  maxIter: Int,
+  periodFeature: TopicCount => Double
+) extends Serializable
+
+object ComputedAnomalyKmeansModelParameter {
+
+  val parametersForAnalysis: Set[ComputedAnomalyKmeansModelParameter] = PeriodType.all.flatMap(kmeansParameters(_, 1, 1))
+
+  val parametersForAnalysisByKey: Map[String, ComputedAnomalyKmeansModelParameter] = parametersForAnalysis.groupBy(_.key).mapValues(v=>v.head)
+
+  def kmeansParameters(periodType: PeriodType,
+                       uniqueSourcesCount: Int,
+                       uniqueTopicsCount: Int): Seq[ComputedAnomalyKmeansModelParameter] = {
+    periodType match {
+      case PeriodType.Minute => Seq(
+        ComputedAnomalyKmeansModelParameter(
+          ComputedAnomalyKmeansModelKey.Minute,
+          60 * uniqueSourcesCount * uniqueTopicsCount, /* base of 60 buckets representing each minute within an hour */
+          20,
+          topicCount => {
+            100 * /* spacing between minute features */
```

Comment:
I'm a big non-fan of magic constants, so I'd either consider factoring this into a well-named function or some sort of alternate approach.